packages/amplify-provider-awscloudformation/src/utils/auth-mode-compare.ts

```diff
@@ -0,0 +1,38 @@
+import _ from "lodash";
+
+export function isAuthModeUpdated(options): boolean {
+  const { authConfig, previousAuthConfig } = getAuthConfigForCompare(options);
+  return authConfig && previousAuthConfig && !_.isEqual(authConfig, previousAuthConfig);
+}
+
+function getAuthConfigForCompare(options) {
```

Comment:
might be cleaner to write a custom equals function rather than cloning and mutating the objects: https://lodash.com/docs/4.17.15#isEqualWith